### PR TITLE
adds mocking response from local file which must be under assets folder

### PR DESCRIPTION
**Pull request by emmar313**

### CODE REVIEW

File: wasp/src/main/java/com/orhanobut/wasp/MockFactory.java [hunk 1]
```diff
@@ -17,32 +20,52 @@
 
     private static MockFactory mockFactory;
 
-    private MockFactory() {
+    private Context context;
+
+    private MockFactory(final Context context) {
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/MockFactory.java]: Would be convenient to remove final declaration from method params.

File: wasp/src/main/java/com/orhanobut/wasp/MockFactory.java [hunk 2]
```diff
@@ -17,32 +20,52 @@
...
             return;
         }
 
-        //Create mock object and return
-        MethodInfo methodInfo = waspRequest.getMethodInfo();
-        Type responseType = methodInfo.getResponseObjectType();
-        String responseString = createJsonString(responseType);
+        String responseString = null;
+
+        if (mock.getPath().equals("")) {
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/MockFactory.java]: I would use TextUtils.isEmpty

File: wasp/src/main/java/com/orhanobut/wasp/MockFactory.java [hunk 3]
```diff
@@ -17,32 +20,52 @@
...
+
+        } else {
+
+            try {
+                responseString = IOUtils.readFileFromAssets(context, mock.getPath());
+            } catch (IOException e) {
+                callBack.onError(new WaspError("mock url", "Mocking failed due to file io error.",
+                        WaspError.INVALID_STATUS_CODE));
+            }
+
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/MockFactory.java]: I would remove empty line here

File: wasp/src/main/java/com/orhanobut/wasp/MockFactory.java [hunk 4]
```diff
@@ -17,32 +20,52 @@
...
+            MethodInfo methodInfo = waspRequest.getMethodInfo();
+            Type responseType = methodInfo.getResponseObjectType();
+            responseString = createJsonString(responseType);
+
+        } else {
+
+            try {
+                responseString = IOUtils.readFileFromAssets(context, mock.getPath());
+            } catch (IOException e) {
+                callBack.onError(new WaspError("mock url", "Mocking failed due to file io error.",
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/MockFactory.java]: We can use group formatting here
<pre>
callBack.onError(new WaspError(
          "mock url", 
          "Mocking failed due to file io error.",
          WaspError.INVALID_STATUS_CODE
));
</pre>

File: wasp/src/main/java/com/orhanobut/wasp/WaspMock.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.orhanobut.wasp;
+
+/**
+ * @author Emmar Kardeslik
+ */
+class WaspMock {
+
+    private int statusCode;
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/WaspMock.java]: We can declare the fields as final

File: wasp/src/main/java/com/orhanobut/wasp/MethodInfo.java [hunk 6]
```diff
@@ -85,7 +85,7 @@ private void parseMethodAnnotations() {
 
             if (annotationType == Mock.class) {
                 Mock mock = (Mock) annotation;
-                mockType = mock.value();
+                this.mock = new WaspMock(mock.statusCode(), mock.path());
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/MethodInfo.java]: Maybe we can put a file exists check here, if the path is not empty,null, we can check if it really exists, if not exists regarding to conditions, would be nice to throw null pointer exception to warn the developer.

File: wasp/src/main/java/com/orhanobut/wasp/MockFactory.java [hunk 7]
```diff
@@ -17,32 +20,52 @@
...
-        String responseString = createJsonString(responseType);
+        String responseString = null;
+
+        if (mock.getPath().equals("")) {
+
+            //Create mock object and return
+            MethodInfo methodInfo = waspRequest.getMethodInfo();
+            Type responseType = methodInfo.getResponseObjectType();
+            responseString = createJsonString(responseType);
+
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/MockFactory.java]: I think we can also remove  this empty line.

File: wasp/src/main/java/com/orhanobut/wasp/IOUtils.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        String str;
+
+        while ((str = reader.readLine()) != null) {
+            builder.append(str);
+        }
+
+        reader.close();
+
+        return builder.toString();
+
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/IOUtils.java]: I would remove this empty line as well here. :)

File: wasp/src/main/java/com/orhanobut/wasp/IOUtils.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * @author Emmar Kardeslik
+ */
+public class IOUtils {
+
+    private IOUtils() {
+        //no instance
+    }
+
+    public static String readFileFromAssets(final Context context, final String filePath) throws IOException {
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/IOUtils.java]: I think we don't need final declaration for Context and filePath here. For the consistency I would not use them.

File: wasp/src/main/java/com/orhanobut/wasp/IOUtils.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+        StringBuilder builder = new StringBuilder();
```
Comment (orhanobut) [inline wasp/src/main/java/com/orhanobut/wasp/IOUtils.java]: What do you think if we put a NP control here for context and filepath and if they are null, we can throw an NPE with a convenient message.

**Comment by emmar313** [comment]
Changes applied according to reviews. Please see new commit.

**Pull request merged**